File: 2470-removing-stars-from-a-string/2470-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        stack<char> stack;
        for (auto ch : s) {
            if (ch == '*')
                stack.pop();
            else
                stack.push(ch);
        }

        string res = "";
        while (!stack.empty()) {
            res += stack.top();
            stack.pop();
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

File: 2470-removing-stars-from-a-string/2470-removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:
        stack = []
        for ch in s:
            if stack and ch == '*':
                stack.pop()
            else:
                stack.append(ch)
        
        res = ""
        while stack:
            res = stack.pop() + res
        
        return res
        

        